File: go/array/1170. 比较字符串最小字母出现频次.go
package array

func numSmallerByFrequency(queries []string, words []string) []int {
	var f func(string) int
	f = func(s string) int {
		num := 0
		mStr := byte('z')
		for i := 0; i < len(s); i++ {
			if s[i] < mStr {
				mStr = s[i]
				num = 1
			} else {
				num++
			}
		}
		return num
	}

	var res []int
	for _, v1 := range queries {
		c := 0
		for _, v2 := range words {
			if f(v1) < f(v2) {
				c++
			}
		}
		res = append(res, c)
	}
	return res
}

File: python/数组/1170. 比较字符串最小字母出现频次.py
from typing import List
class Solution:
    def numSmallerByFrequency(self, queries: List[str], words: List[str]) -> List[int]:
        from collections import Counter
        def f(s):
            if not s:
                return 0
            d = dict(Counter(s))
            return sorted(d.items(),key=lambda x:(x[0],[1]))[0][1]
        queries = list(map(f,queries))
        words = list(map(f,words))
        res = []
        for i in queries:
            c = 0
            for j in words:
                if i<j:
                    c+=1
            res.append(c)
        return res
